src/test/ui/rfc-2632-const-trait-impl/const-drop.rs

```diff
@@ -60,7 +60,7 @@ mod t {
         fn foo() {}
     }
 
-    pub struct ConstDropWithBound<T: SomeTrait>(pub core::marker::PhantomData<T>);
+    pub struct ConstDropWithBound<T: ~const SomeTrait>(pub core::marker::PhantomData<T>);
```

Comment:
what does `~const Trait` mean in a struct definition? I don't feel like "`const` if `const`" makes sense for types

Reply:
Yea... this is somewhat borked... we require a drop impl to have the exact same bounds as the struct, and I don't see how to do this other than adding an exception for constness.

Reply:
It suggests that "For `SomeType` to be WF at a const use site, `T` must implement `const SomeTrait`". As oli said, this is needed when we call `<T as ~const SomeTrait>::foo` in the destructor impl, otherwise it would be impossible to write a compile-time destructor that calls a trait method.